src/setup_payload/tests/TestAdditionalDataPayload.cpp

```diff
@@ -0,0 +1,197 @@
+}
+
+void TestGeneratingAdditionalDataPayloadWithoutRotatingId(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, ArraySize(serialNumber),
+                                                                         bufferHandle, additionalDataFields);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, !bufferHandle.IsNull());
+
+    char output[51];
+    err = BytesToUppercaseHexString(bufferHandle->Start(), bufferHandle->DataLength(), output, ArraySize(output));
+
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(output, "55000018") == 0);
+}
+
+void TestGeneratingAdditionalDataPayloadWithRotatingIdAndParsingIt(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+    additionalDataFields.Set(AdditionalDataFields::RotatingDeviceId);
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, ArraySize(serialNumber),
+                                                                         bufferHandle, additionalDataFields);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, !bufferHandle.IsNull());
+
+    char output[51];
+    err = BytesToUppercaseHexString(bufferHandle->Start(), bufferHandle->DataLength(), output, ArraySize(output));
+
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(output, "5500003000120A00D1414E39C9A08469DA2E29F0ACAA086618") == 0);
+
+    AdditionalDataPayload resultPayload;
+    err = AdditionalDataPayloadParser(bufferHandle->Start(), bufferHandle->DataLength()).populatePayload(resultPayload);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(resultPayload.rotatingDeviceId.c_str(), "0A00D1414E39C9A08469DA2E29F0ACAA0866") == 0);
+}
+
+void TestGeneratingAdditionalDataPayloadWithoutRotatingIdAndParsingIt(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
```

Comment:
This can be done in a Setup/Teardown combo (see other unit tests that call MemoryInit()).

Reply:
i agree, updated